test/expected/alter.out

```diff
@@ -734,6 +734,35 @@ SELECT hypertable_constraint_name, constraint_name from _timescaledb_catalog.chu
 
 INSERT INTO t_hypertable AS h VALUES ( 1, '2020-01-01 00:01:00', 3.2) ON CONFLICT (id, time) DO UPDATE SET value = h.value + EXCLUDED.value;
 ROLLBACK;
+-- predicate reconstruction when attnos are different in hypertable and chunk
+CREATE TABLE p_hypertable (a timestamp, b integer, c integer);
+SELECT create_hypertable('p_hypertable', 'a', chunk_time_interval => interval '3 day');
+NOTICE:  adding not-null constraint to column "a"
+     create_hypertable      
+----------------------------
+ (14,public,p_hypertable,t)
+(1 row)
+
+BEGIN;
+ALTER TABLE p_hypertable DROP COLUMN b, ADD COLUMN d boolean;
+CREATE INDEX idx_ht ON p_hypertable(a, c) WHERE d = FALSE;
+END;
+INSERT INTO p_hypertable(a, c, d) VALUES (now(), 1, FALSE);
+\d+ _timescaledb_internal._hyper_14_28_chunk
+                               Table "_timescaledb_internal._hyper_14_28_chunk"
+ Column |            Type             | Collation | Nullable | Default | Storage | Stats target | Description 
+--------+-----------------------------+-----------+----------+---------+---------+--------------+-------------
+ a      | timestamp without time zone |           | not null |         | plain   |              | 
+ c      | integer                     |           |          |         | plain   |              | 
+ d      | boolean                     |           |          |         | plain   |              | 
+Indexes:
+    "_hyper_14_28_chunk_idx_ht" btree (a, c) WHERE NOT d
+    "_hyper_14_28_chunk_p_hypertable_a_idx" btree (a DESC)
+Check constraints:
+    "constraint_34" CHECK (a >= 'Mon Jan 10 00:00:00 2022'::timestamp without time zone AND a < 'Thu Jan 13 00:00:00 2022'::timestamp without time zone)
```

Comment:
I believe this output will keep on changing with time. "Jan 10 to Jan 13". Better to use integer `chunk_time_interval` value like `chunk_time_interval => 100000` as an example.